### PR TITLE
Prepare `v10.0.0-beta.0` for release

### PR DESCRIPTION
**Pull request by Shaptic**
This is the final PR into the `soroban` branch, preparing a new major version (though tagged with `@beta`) for release to NPM: v10.0.0-beta.0.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,8 +3,37 @@
...
 
-### Fixes
 
+## [`v10.0.0-beta.0`](https://github.com/stellar/js-stellar-base/compare/v9.0.0...v10.0.0-beta.0): Protocol 20
+
+### Breaking Changes
+ * **Node 16 is the new minimum version** to use the SDKs.
+ * The XDR has been massively overhauled to support Soroban in Protocol 20, which means new operations, data structures, and a transaction format as well as new overlay features ([#538](https://github.com/stellar/js-stellar-base/pull/538)).
+
+The core data structure of Soroban is a generic type called an `ScVal` (**s**mart **c**ontract **val**ue, which is a union of types that can basically represent anything [numbers, strings, arrays, maps, contract bytecode, etc.]). You can refer to the XDR for details, and you can rely some new APIs to make dealing with these complex values easier:
```
Comment (sreuland) [inline CHANGELOG.md:13]:
```suggestion
The core data structure of Soroban is a generic type called an `ScVal` (**s**mart **c**ontract **val**ue, which is a union of types that can basically represent anything [numbers, strings, arrays, maps, contract bytecode, etc.]). You can refer to the XDR for details, and you can utilize new APIs to make dealing with these complex values easier:
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,8 +3,37 @@
...
+ - `nativeToScVal` helps convert native types to their closes Soroban equivalent
```
Comment (sreuland) [inline CHANGELOG.md:14]:
```suggestion
 - `nativeToScVal` helps convert native types to their closest Soroban equivalent
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,8 +3,37 @@
 
 ## Unreleased
 
-### Fixes
 
+## [`v10.0.0-beta.0`](https://github.com/stellar/js-stellar-base/compare/v9.0.0...v10.0.0-beta.0): Protocol 20
+
+### Breaking Changes
+ * **Node 16 is the new minimum version** to use the SDKs.
+ * The XDR has been massively overhauled to support Soroban in Protocol 20, which means new operations, data structures, and a transaction format as well as new overlay features ([#538](https://github.com/stellar/js-stellar-base/pull/538)).
```
Comment (sreuland) [inline CHANGELOG.md:11]:
```suggestion
 * The XDR has been massively overhauled to support [Soroban in Protocol 20](https://soroban.stellar.org/docs/category/fundamentals-and-concepts), which means new operations, data structures, and a transaction format as well as new overlay features ([#538](https://github.com/stellar/js-stellar-base/pull/538)).
```

**Review by sreuland — APPROVED** [review]
outstanding!

File: CHANGELOG.md (hunk 3 above)
Comment (Shaptic) [reply CHANGELOG.md]: great idea!

**Pull request merged**